compiler/rustc_passes/src/check_attr.rs

```diff
@@ -375,7 +375,7 @@ impl<'tcx> CheckAttrVisitor<'tcx> {
         );
     }
 
-    /// Checks if `#[diagnostic::do_not_recommend]` is applied on a trait impl.
+    /// Checks if `#[diagnostic::do_not_recommend]` is applied on a trait impl and that it is a word.
```

Comment:
```suggestion
    /// Checks if `#[diagnostic::do_not_recommend]` is applied on a trait impl
    /// and that it is a word.
```

Wrapping.  Also, perhaps for the purposes of the doc comment, it might be worth inlining what we mean by a "word".

Reply:
The phrasing is now "and that it has no arguments".